mypy/util.py

```diff
@@ -467,43 +467,66 @@ class FancyFormatter:
     def __init__(self, f_out: IO[str], f_err: IO[str], show_error_codes: bool) -> None:
         self.show_error_codes = show_error_codes
         # Check if we are in a human-facing terminal on a supported platform.
-        if sys.platform not in ('linux', 'darwin'):
+        if sys.platform not in ('linux', 'darwin', 'win32'):
             self.dummy_term = True
             return
         force_color = int(os.getenv('MYPY_FORCE_COLOR', '0'))
         if not force_color and (not f_out.isatty() or not f_err.isatty()):
             self.dummy_term = True
             return
-
-        # We in a human-facing terminal, check if it supports enough styling.
-        if not CURSES_ENABLED:
-            self.dummy_term = True
-            return
-        try:
-            curses.setupterm()
-        except curses.error:
-            # Most likely terminfo not found.
-            self.dummy_term = True
-            return
-        bold = curses.tigetstr('bold')
-        under = curses.tigetstr('smul')
-        set_color = curses.tigetstr('setaf')
-        self.dummy_term = not (bold and under and set_color)
-        if self.dummy_term:
-            return
-
-        self.NORMAL = curses.tigetstr('sgr0').decode()
-        self.BOLD = bold.decode()
-        self.UNDER = under.decode()
-        dim = curses.tigetstr('dim')
-        # TODO: more reliable way to get gray color good for both dark and light schemes.
-        self.DIM = dim.decode() if dim else PLAIN_ANSI_DIM
-
-        self.BLUE = curses.tparm(set_color, curses.COLOR_BLUE).decode()
-        self.GREEN = curses.tparm(set_color, curses.COLOR_GREEN).decode()
-        self.RED = curses.tparm(set_color, curses.COLOR_RED).decode()
-        self.YELLOW = curses.tparm(set_color, curses.COLOR_YELLOW).decode()
-
+        if sys.platform == 'win32':
+            # Windows ANSI escape sequences are only supported on Threshold 2 and above.
+            if sys.getwindowsversion().build < 10586:
```

Comment:
I would move this constant (10586) to the top of file, and add a bit more info about this.